src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -627,18 +632,22 @@ public function getEditLink($id)
      */
     private function getGridFieldItemAdjacencies(): array
     {
-        $list = $this->getGridField()->getManipulatedList();
-        $paginator = $this->getGridFieldPaginatorState();
-        if (!$paginator) {
-            return [];
+        if (!isset($this->cachedGridFieldItemAdjacencies)) {
+            $paginator = $this->getGridFieldPaginatorState();
+            if (!$paginator) {
+                $this->cachedGridFieldItemAdjacencies = [];
+            } else {
+                $currentPage = $paginator->getData('currentPage');
+                $itemsPerPage = $paginator->getData('itemsPerPage');
+                $limit = $itemsPerPage + 2;
+                $limitOffset = max(0, $itemsPerPage * ($currentPage - 1) -1);
```

Comment:
```suggestion
                $limitOffset = max(0, $itemsPerPage * ($currentPage - 1) - 1);
```
Since we're adding space around the other `-` we should do it to this one.

Reply:
Updated